src/NHibernate/Hql/Ast/ANTLR/Tree/EntityJoinJoinSequenceImpl.cs

```diff
@@ -8,44 +7,9 @@ namespace NHibernate.Hql.Ast.ANTLR.Tree
 {
 	class EntityJoinJoinSequenceImpl : JoinSequence
```

Comment:
Let's just kill this class.

Reply:
I thought to keep it just for reference with hibernate implementation. 

>Is it a backport of some changes from master

Yes it's been simplified with table group joins implementation https://github.com/nhibernate/nhibernate-core/pull/2361

Reply:
> Yes it's been simplified with table group joins implementation

I don't get it. In 5.3.x we don't have table group joins, but the filters still work?

 > I thought to keep it just for reference with hibernate implementation.

Ok. Let's don't keep it.

Reply:
> In 5.3.x we don't have table group joins, but the filters still work?

Yep. Whatever is supported for many-to-one is now supported for entity join. We just removed custom entity join processing

Reply:
>Ok. Let's don't keep it.

Ok But let's at least have comment about it.